wcfsetup/install/files/lib/data/user/avatar/UserAvatarEditor.class.php

```diff
@@ -89,46 +90,29 @@ public function createAvatarVariant(): bool {
 			return false;
 		}
 		
-		// The image adapters do not support the conversion to different
-		// file types. However, we do require full GD support for jpeg, png
-		// and WebP, therefore we can safely rely on the GD library for
-		// the conversion. The images are so small, that neither the speed
-		// nor the quality will noticeably differ from Imagick.
 		$filename = $this->getLocation();
 		$filenameWebP = $this->getLocation(null, true);
 		
+		$imageAdapter = ImageHandler::getInstance()->getAdapter();
+		$imageAdapter->loadFile($filename);
+		$image = $imageAdapter->getImage();
+		
 		$data = ["hasWebP" => 1];
 		
 		// If the uploaded avatar is already a WebP image, then create a JPEG
 		// as a fallback image and flip the image data to match the JPEG.
 		if ($this->avatarExtension === "webp") {
 			$filenameJpeg = preg_replace('~\.webp$~', '.jpeg', $filenameWebP);
 			
-			$source = imagecreatefromwebp($filename);
-			imagejpeg($source, $filenameJpeg);
+			$imageAdapter->saveImageAs($image, $filenameJpeg, "jpeg", 80);
 			
 			$data = [
 				"avatarExtension" => "jpeg",
 				"fileHash" => sha1_file($filenameJpeg),
 			];
 		}
 		else {
-			// The source avatar is not a WebP image.
-			switch ($this->avatarExtension) {
-				case "jpg":
-				case "jpeg":
-					$source = imagecreatefromjpeg($filename);
-					break;
-				
-				case "png":
-					$source = imagecreatefrompng($filename);
-					break;
-				
-				default:
-					throw new \LogicException("Unreachable");
-			}
-			
-			imagewebp($source, $this->getLocation(null, true));
+			$imageAdapter->saveImageAs($imageAdapter, $this->getLocation(null, true), "webp", 80);
```

Comment:
`$imageAdapter->saveImageAs($image, $this->getLocation(null, true), "webp", 80);`

Reply:
Good catch, thanks!